.github/workflows/syntax_check_cloud.yml

```diff
@@ -0,0 +1,31 @@
+# This workflow will check for exceptions to raise properly and do not generate syntax errors.
+# Also old style classes with \"object\" as parent will be checked
+
+name: Syntax checking
+
+on:
+  [push, pull_request]
+jobs:
+  build:
+
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-20.04, ubuntu-22.04]
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Check exceptions
+      # Not related to tests, but to QA in general: Exceptions usually take only 1 argument
+      # So a comma is probably a sign of syntax error and should be replaced by a %
+      run: |
+        export ODEMIS_DIR=.
+        export ODEMIS_SRC="$ODEMIS_DIR/src/odemis"
+        grep -IrE --colour 'raise .+".*%.*",' --include=*.py "$ODEMIS_SRC" "$ODEMIS_DIR"/scripts/ "$ODEMIS_DIR"/plugins/      
```

Comment:
grep returns 0 if it finds nothing and 1 if it finds something.
Doesn't it work if it's the last command run?
Otherwise you could just write `exit $?`